app/components/orkestrator-barn/OrkestratorBarn.tsx

```diff
@@ -0,0 +1,160 @@
+import { Button, Heading, Modal } from "@navikt/ds-react";
+import { useActionData, useNavigation } from "@remix-run/react";
+import { useForm } from "@rvf/remix";
+import { withZod } from "@rvf/zod";
+import classNames from "classnames";
+import { useEffect, useRef } from "react";
+import { z } from "zod";
+
+import { useTypedRouteLoaderData } from "~/hooks/useTypedRouteLoaderData";
+import { IOrkestratorBarn } from "~/models/orkestrator-opplysning.server";
+import { action } from "~/routes/oppgave.$oppgaveId.behandle";
+
+import styles from "./OrkestratorBarn.module.css";
+import { OrkestratorOpplysningLinje } from "./OrkestratorOpplysningLinje";
+
+interface IProps {
+  barnNummer: number;
+  barn: IOrkestratorBarn;
+}
+
+const validator = withZod(
+  z.object({
+    fornavnOgMellomnavn: z.string().min(1, { message: "Du må skrive fornavn" }),
+    etternavn: z.string().min(1, { message: "Du må skrive etternavn" }),
+    fødselsdato: z.string().regex(
+      new RegExp("^(0[1-9]|[12][0-9]|3[01])[\\.-](0[1-9]|1[012])[\\.-](19|20|)\\d\\d$"), // Regex for å matche norsk dato format, eks. 01.02.2023
+      "Ugyldig dato. Gylige datoformat er dd.mm.åååå",
+    ),
+    oppholdssted: z.string().min(1, { message: "Du må velge et land" }),
+    forsørgerBarnet: z.enum(["true", "false"], {
+      required_error: "Du må velge et svar",
+      invalid_type_error: "Ugyldig svar",
+    }),
+    kvalifisererTilBarnetillegg: z.enum(["true", "false"], {
+      required_error: "Du må velge et svar",
+      invalid_type_error: "Ugyldig svar",
+    }),
+    barnetilleggFom: z
+      .string()
+      .regex(
+        new RegExp("^(0[1-9]|[12][0-9]|3[01])[\\.-](0[1-9]|1[012])[\\.-](19|20|)\\d\\d$"),
+        "Ugyldig dato. Gylige datoformat er dd.mm.åååå",
+      ),
+    barnetilleggTom: z
+      .string()
+      .regex(
+        new RegExp("^(0[1-9]|[12][0-9]|3[01])[\\.-](0[1-9]|1[012])[\\.-](19|20|)\\d\\d$"),
+        "Ugyldig dato. Gylige datoformat er dd.mm.åååå",
+      ),
+    begrunnelse: z.string().min(1, { message: "Du må skrive begrunnelse" }),
+  }),
+);
+
+export function OrkestratorBarn({ barnNummer, barn }: IProps) {
+  const ref = useRef<HTMLDialogElement>(null);
+  const actionData = useActionData<typeof action>();
+  const { state } = useNavigation();
+  const { oppgave } = useTypedRouteLoaderData("routes/oppgave.$oppgaveId");
+
+  useEffect(() => {
+    if (actionData) {
+      ref.current?.close();
+    }
+  }, [actionData]);
+
+  const defaultValues = barn.opplysninger.reduce(
+    (acc, { id, verdi }) => {
+      acc[id] = verdi;
+      return acc;
+    },
+    {} as Record<string, string>,
+  );
+
+  const orkestratorBarnForm = useForm({
+    validator: validator,
+    method: "put",
+    defaultValues: defaultValues,
+  });
+
+  function avbryt() {
+    orkestratorBarnForm.resetForm();
+    ref.current?.close();
+  }
+
+  // Todo: Ikke lagre til backend dersom ingenting er endret i formen
```

Comment:
Bruk:
orkestratorBarnForm.formState.isDirty for å shekke om man skal lagre eller ikke